### PR TITLE
22: Add PostCSS

### PR DESCRIPTION
**Pull request by katiefenn**
- Added PostCSS pre-processor
- Added postcss-import plugin
- Added postcss-simple-vars plugin

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -50,9 +50,10 @@
     "style-loader": "^0.13.1",
     "webpack": "^1.13.1",
     "webpack-dev-middleware": "^1.6.1",
-    "webpack-hot-middleware": "^2.12.2"
-  },
-  "devDependencies": {
-    "extract-text-webpack-plugin": "^1.0.1"
```
Comment (katiefenn) [inline package.json]: Moved all the things plugged into WebPack into `dependencies` instead of `devDependencies`

File: webpack/dev-config.js [hunk 2]
```diff
@@ -1,6 +1,8 @@
 const path = require('path');
 const webpack = require('webpack');
 const ExtractTextPlugin = require('extract-text-webpack-plugin');
+const atImport = require('postcss-import');
+const variables = require('postcss-simple-vars');
```
Comment (katiefenn) [inline webpack/dev-config.js]: Added PostCSS plugins for Sass-esque loading of files and variables.

File: webpack/dev-config.js [hunk 3]
```diff
@@ -28,10 +30,32 @@ module.exports = {
...
+          ].join('!'),
+          [
+            loader(
+              'css-loader',
+              'modules',
+              'importLoaders=1',
+              'localIdentName=[name]__[local]___[hash:base64:5]'
+            ),
+            'postcss-loader'
+          ].join('!')
```
Comment (katiefenn) [inline webpack/dev-config.js]: Tried to make config more readable than the rubbish WebPack loader config strings.

Shout if this is over-the-top.

File: webpack/dev-config.js [hunk 4]
```diff
@@ -28,10 +30,32 @@ module.exports = {
...
         )
       }
     ]
+  },
+  postcss: function () {
+    return [atImport, variables];
   }
 };
+
```
Comment (katiefenn) [inline webpack/dev-config.js]: Helper function for making WebPack loader config strings.

File: package.json (hunk 1 above)
Comment (earnubs) [inline package.json]: since we'll never publish this as a module, does it matter? (and then, why not have everything in deps )

Comment (katiefenn) [inline package.json]: On the last app I worked on we bundled our `dependencies` with the apps in a single JS file (excluding `devDependencies`). Moving mocha to devDependencies would reduce the response by 9K (unzipped) in filesize in this case.

Perhaps the best thing to do would be bung everything in `dependencies` and review when we actually intend to deploy?

**Pull request closed without merging**